src/test/java/com/github/checkstyle/patchfilter/SuppressionPatchFilterTest.java

```diff
@@ -51,33 +58,166 @@ public void testAccept() throws Exception {
-        final LocalizedMessage message3 = new LocalizedMessage(7, 1, null, "msg", null,
-                SeverityLevel.ERROR, null, getClass(), null);
-        final AuditEvent ev3 = new AuditEvent(this, "Test1.java", message1);
-        assertTrue(filter.accept(ev1),
-                "Audit event should be rejected when there are no matching patch filters");
+        final FileInputStream ins = new FileInputStream(fileName);
+        final UnifiedDiff diff = UnifiedDiffReader.parseUnifiedDiff(ins);
+        assertEquals(2, diff.getFiles().size());
+        final UnifiedDiffFile file1 = diff.getFiles().get(0);
+        assertEquals("diff -u Origin/Test1.java Update/Test1.java",
+                file1.getDiffCommand());
+        assertEquals("Origin/Test1.java", file1.getFromFile());
+        assertEquals("Update/Test1.java", file1.getToFile());
+        assertEquals(1, file1.getPatch().getDeltas().size());
+        final UnifiedDiffFile file2 = diff.getFiles().get(1);
+        assertEquals("diff -u Origin/Test2.java Update/Test2.java",
+                file2.getDiffCommand());
+        assertEquals("Origin/Test2.java", file2.getFromFile());
+        assertEquals("Update/Test2.java", file2.getToFile());
+        assertEquals(1, file2.getPatch().getDeltas().size());
+        assertNull(diff.getTail());
+    }
+
+    @Test
+    public void testGitDiffPatch() throws Exception {
+        final String fileName = getPath("PatchFileFromDiffTools/GitDiffPatch.txt");
+        final FileInputStream ins = new FileInputStream(fileName);
+        final UnifiedDiff diff = UnifiedDiffReader.parseUnifiedDiff(ins);
+        assertEquals(1, diff.getFiles().size());
+        final UnifiedDiffFile file1 = diff.getFiles().get(0);
+        assertEquals("diff --git a/pom.xml b/pom.xml", file1.getDiffCommand());
+        assertEquals("pom.xml", file1.getFromFile());
+        assertEquals("pom.xml", file1.getToFile());
+        assertEquals(1, file1.getPatch().getDeltas().size());
+        assertNull(diff.getTail());
+    }
+
+    @Test
+    public void testGitFormatPatchLinuxOne() throws Exception {
+        final String fileName = getPath("PatchFileFromDiffTools/GitFormatPatchLinuxOne.txt");
+        final FileInputStream ins = new FileInputStream(fileName);
+        final UnifiedDiff diff = UnifiedDiffReader.parseUnifiedDiff(ins);
+        assertEquals(1, diff.getFiles().size());
+        final UnifiedDiffFile file1 = diff.getFiles().get(0);
+        assertEquals("diff --git a/pom.xml b/pom.xml", file1.getDiffCommand());
+        assertEquals("pom.xml", file1.getFromFile());
+        assertEquals("pom.xml", file1.getToFile());
+        assertEquals(1, file1.getPatch().getDeltas().size());
+        assertEquals("2.7.4\n\n", diff.getTail());
+    }
+
+    @Test
+    @Ignore("error: com.github.difflib.unifieddiff.UnifiedDiffParserException: "
+            + "expected file start line not found")
```

Comment:
Here we need information about when we can activate this test. Please put here a link to an issue in the library which will fix this problem. Or create new issue if it doesn't exist

Reply:
done, I make a issue to java-diff-utils add "until https://github.com/java-diff-utils/java-diff-utils/issues/84"